arch/arm/src/samv7/sam_mcan.c

```diff
@@ -4380,14 +4381,10 @@ struct can_dev_s *sam_mcan_initialize(int port)
 
       /* Get the revision of the chip (A or B) */
 
-#ifdef CONFIG_ARCH_CHIP_PIC32CZCA70
-      /* PIC32CZ CA70 series always have revision B MCAN */
-
-      priv->rev = 1;
-#else
-      regval = getreg32(SAM_CHIPID_CIDR);
-      priv->rev = regval & CHIPID_CIDR_VERSION_MASK;
-#endif
+      if (sam_has_revb_periphs())
+        {
+          priv->rev = 1;
```

Comment:
This doesn't mirror the ifdef logic it replaces, isn't rev now undefined if it's not REV A?

Reply:
I see, I'm not familiar with the contents of `SAM_CHIPID_CIDR` but I assumed the version could be multiple non-zero things.